scripts_src/newreno/nclabish.ssl

```diff
@@ -166,6 +166,22 @@ procedure Node999;                                      // This Node is Always E
 
 #define mrs_bishop_in_bed                 ((self_tile == mrs_bishop_self_bed_tile) and (self_elevation == mrs_bishop_room_elev))
 
+#define th_start_dialog_at_node(x)                       if (in_dialog == false) then begin           \
+                                                         in_dialog := true;                        \
+                                                         start_gdialog(NAME,self_obj,4,HEAD_LBSHP,BACKGROUND_RAIDERS);     \
+                                                         gSay_Start;                               \
+                                                            call x;                                \
+                                                         gSay_End;                                 \
+                                                         end_dialogue;                             \
+                                                         in_dialog := false;                       \
+                                                         end else call x
+
```

Comment:
Again, we may want to move these custom functions out of the SSL to some where that they are shared, so if we need to make an update to the function call, we only have to do it in one place.

Reply:
I can redo all these scripts with a new universal macro in the **define.h** if you wish